client/src/main/scala/com.gu.contentapi.client/GuardianContentClient.scala

```diff
@@ -107,26 +107,38 @@ trait ContentApiClientLogic {
       }
     }
 
-  /* Exposed API */
-
+  /** Runs the query against the Content API.
+    * 
+    * @tparam Q the type of a Content API query
+    * @param query the query
+    * @return a future resolving to an unmarshalled response
+    */
   def getResponse[Q <: ContentApiQuery](query: Q)(
     implicit 
     decoder: Decoder[Q],
-    context: ExecutionContext): Future[decoder.R] =
+    context: ExecutionContext): Future[decoder.Response] =
     fetchResponse(query) map decoder.decode
 
-  def paginate[Q <: PaginatedApiQuery[Q], RR: MetaResult](q: Q)(f: RR => Future[Unit])(
+  /** Runs a query and process all the pages of results.
+    * 
+    * @tparam Q the type of a Content API query with paging parameters
```

Comment:
[trivial] We should probably use the word "pagination" instead of "paging" as they can have subtly different meanings.